qoffer-domain/src/main/groovy/life/qbic/portal/portlet/offers/Offer.groovy

```diff
@@ -115,11 +133,52 @@ class Offer {
      * @return The total costs in the currency of the selected items.
      */
     double getTotalCosts() {
-        final double netPrice = calculateNetPrice()
-        final double overhead = determineOverhead()
-        final double vat = determineVat()
-        // TODO check back with BioPM if this is correct
-        return netPrice*overhead + vat*netPrice
+        calculateTotalCosts()
+    }
+
+    /**
+     * The total net price for the current offer.
+     *
+     * Note: Does <strong>not include</strong> overheads and taxes.
+     *
+     * @return The net offer price
+     */
+    double getTotalNetPrice() {
+        return calculateNetPrice()
+    }
+
+    /**
+     * The overhead price amount of all service items without VAT.
+     *
+     * Service items of type data storage and project management
+     * are <strong>excluded</strong> from he calculation.
+     *
+     * @return The calculated overhead amount of the selected items.
+     */
+    double getOverheadSum() {
+        double overheadSum = 0
+        for (ProductItem item : items) {
+            if (item.product instanceof DataStorage || item.product instanceof ProjectManagement) {
+                // No overheads are assigned for data storage and project management
+            } else {
+                overheadSum += item.quantity * item.product.unitPrice * this.overhead
+            }
+        }
+        return overheadSum
+    }
+
+    /**
+     * The tax price on all items net price including overheads.
+     *
+     * For internal customers, this will be 0.
+     *
+     * @return The amount of VAT price based on all items in the offer.
+     */
+    double getVatSum() {
```

Comment:
I find that the term `VatSum` is hard to understand. For me it would be easier to read to have a method `double withVatApplied(double priceWithoutVat)` or `double applyVat(doubli priceWithoutVat)`.

That would also make it easier to understand what happens in your `calculateTotalCosts` method and the `calculateNetPrice` and `getOverheadSum` would only need to be called once there. 

Since this class does not implement any interface or abstract method I guess you are still flexible with the API right now.